src/app/dashboard-layout/courses-list/add-course/creation-course-date/creation-course-date.component.ts

```diff
@@ -17,45 +22,96 @@ import { FormBuilder, FormGroup } from "@angular/forms";
   templateUrl: "./creation-course-date.component.html",
   styleUrls: ["./creation-course-date.component.scss"],
   changeDetection: ChangeDetectionStrategy.OnPush,
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => CreationCourseDateComponent),
+      multi: true,
+    },
+    {
+      provide: NG_VALIDATORS,
+      useExisting: forwardRef(() => CreationCourseDateComponent),
+      multi: true,
+    },
+  ],
 })
-export class CreationCourseDateComponent implements OnInit {
+export class CreationCourseDateComponent implements OnInit, ControlValueAccessor, Validator {
   private fb: FormBuilder;
+  private onTouched: () => void;
+  private onChange: (_: any) => void;
 
   /**
    * Form group
    */
-  public courseForm: FormGroup;
+  public dateCourseForm: FormGroup;
+
+  constructor(@Inject(FormBuilder) fb: FormBuilder) {
+    this.fb = fb;
+    this.onTouched = () => {};
+    this.onChange = (_: any) => {};
+  }
 
   /**
-   * serializedDate
+   * ngOnInit
    */
-  @Input()
-  public dateValue: string;
+  public ngOnInit(): void {
+    this.dateCourseForm = this.fb.group({
```

Comment:
you are creating form group just for one control. Why don't you use createControl instead?